### PR TITLE
Get rid amavisd/spamassassin and use rspamd

### PR DESCRIPTION
**Pull request by stephdl**
First PR to get rid of amavisd/spamassassin in nethserver-mail-filter.
I have to implement the rewrite subject in rspamd, I raise a bug in gh about this. WIP

If your settings about greylisting, tag spam and reject spam are not applied, think to check the map /var/lib/rspamd/rspamd_dynamic in the configuration menu of rspamd, it is made to override values set by configuration files.

black/list about domain or address and file extension rejected can be set in the server manager or in a map (configuration menu of rspamd)

Please test and comment

NethServer/dev#5394

### CODE REVIEW

File: nethserver-mail-filter.spec [hunk 1]
```diff
@@ -1,64 +1,56 @@
...
 License: GPL
 URL: %{url_prefix}/%{name} 
 Source0: %{name}-%{version}.tar.gz
 BuildArch: noarch
-%define policyd_spf_dir postfix-policyd-spf-perl-2.010
-
-Requires: nethserver-mail-common, nethserver-antivirus
+Requires: rspamd
+Requires: nethserver-antivirus
+Requires: nethserver-mail-server
```
Comment (DavidePrincipi) [inline nethserver-mail-filter.spec]: The mail-filter component cannot require the mail-server part to allow an installation that works as SMTP filter only. We must revert to `Requires: nethserver-mail-common, nethserver-antivirus`

File: nethserver-mail-filter.spec [hunk 2]
```diff
@@ -1,64 +1,56 @@
...
 
 BuildRequires: perl
-BuildRequires: nethserver-devtools
+BuildRequires: nethserver-devtools 
 
 %description
-Configures postfix to filter SMTP sessions and mail messages through 
-- amavisd-new (spamassassin, clamav)
-- Mail::SPF (optional)
+Rspamd is an advanced spam filtering system that allows evaluation of messages
```
Comment (DavidePrincipi) [inline nethserver-mail-filter.spec]: "Configures rspamd that is..."

File: nethserver-mail-filter.spec [hunk 3]
```diff
@@ -1,64 +1,56 @@
...
+mkdir -p root/var/run/clamd.rspamd
+mkdir -p root/var/lib/redis/rspamd
+mkdir -p root/var/run/redis-rspamd
 (cd root; find . -depth -print | cpio -dump %{buildroot})
-%{genfilelist} %{buildroot} > %{name}-%{version}-filelist
-echo "%docdir $RPM_DOC_DIR/${POLICYD_SPF_DIR}" >> %{name}-%{version}-filelist
+%{genfilelist} %{buildroot} \
+  --dir /var/run/clamd.rspamd 'attr(0750,_rspamd,_rspamd)' \
+  --dir /var/lib/redis/rspamd 'attr(0755,redis,redis)' \
+  --dir /var/run/redis-rspamd 'attr(0755,redis,redis)' \
```
Comment (DavidePrincipi) [inline nethserver-mail-filter.spec]: Can we mark it %ghost, or not declare it at all, with the `RuntimeDirectory` in `redis-rspamd.service`?

File: nethserver-mail-filter.spec [hunk 4]
```diff
@@ -1,64 +1,56 @@
...
+> %{name}-%{version}-filelist
+echo "%doc COPYING" >> %{name}-%{version}-filelist
```
Comment (DavidePrincipi) [inline nethserver-mail-filter.spec]: Please leave it under the `%files` section

File: nethserver-mail-filter.spec [hunk 5]
```diff
@@ -1,64 +1,56 @@
...
+
+%post
+/usr/bin/systemctl daemon-reload
+%preun
 
 %files -f %{name}-%{version}-filelist
 %defattr(-,root,root)
-%doc COPYING
-%doc README.rst
```
Comment (DavidePrincipi) [inline nethserver-mail-filter.spec]: Please do not remove the documentation file. We'll fix it with the new features/props :)

File: nethserver-mail-filter.spec [hunk 6]
```diff
@@ -1,64 +1,56 @@
...
 %dir %{_nseventsdir}/%{name}-update
 
+%changelog
+
```
Comment (DavidePrincipi) [inline nethserver-mail-filter.spec]: Avoid whitespace changes that clutter the diff

File: root/etc/e-smith/db/configuration/migrate/MigrateAmavisdPropertyToRspamd [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/migrate/MigrateAmavisdPropertyToRspamd]: An unwritten naming convention for migrate fragments is `issueXXXX.ns7`: the migrate fragment will be removed on ns8. Add more info in a comment block here.

File: root/etc/redis-rspamd.conf [hunk 8]
```diff
@@ -0,0 +1,57 @@
+bind 127.0.0.1
```
Comment (DavidePrincipi) [inline root/etc/redis-rspamd.conf]: is it still required if `unixsocket` is also specified?

**Review by DavidePrincipi — CHANGES_REQUESTED** [review]
- Do redis and rspamd need a pre-backup script to dump  their database under `/var/lib`?
- As /run is a tmpfs the /var/run/redis-rspamd directory must be created automatically at every system boot by either `systemd-tmpfiles` or (preferred) `RuntimeDirectory` systemd directive (see redis.service for instance)

File: root/etc/redis-rspamd.conf (hunk 8 above)
Comment (stephdl) [reply root/etc/redis-rspamd.conf]: when commented, do systemctl status

Main PID: 11731 (redis-server)
CGroup: /system.slice/redis-rspamd.service
           └─11731 /usr/bin/redis-server *:0

or if not commented

Main PID: 11778 (redis-server)
   CGroup: /system.slice/redis-rspamd.service
           └─11778 /usr/bin/redis-server 127.0.0.1:0

I do prefer to bind to localhost, eventy if `port 0` disable TCP socket

File: root/etc/e-smith/templates/etc/rspamd/local.d/worker-normal.inc/10base [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# socket for worker-normal
+bind_socket = "/var/run/rspamd/worker-normal mode=0770 owner=_rspamd group=mail";
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/rspamd/local.d/worker-normal.inc/10base]: The documentation says to disable worker-normal in self-scan mode: does it apply here?

https://rspamd.com/doc/workers/rspamd_proxy.html#self-scan-mode

File: root/usr/share/nethesis/NethServer/Module/Dashboard/Applications/Rspamd.php [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        return "Rspamd";
+    }
+
+    public function getInfo()
+    {
+        $webapp = $this->getPlatform()->getDatabase('configuration')->getProp('rspamd','Name');
+        $host = explode(':',$_SERVER['HTTP_HOST']);
+         return array(
+            'url' => "https://".$host[0]."/$webapp/",
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/Applications/Rspamd.php]: Point to the instance on port 980

File: root/usr/share/nethesis/NethServer/Module/Dashboard/Applications/Rspamd.php [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+ * Rspamd interface
+ *
+ * @author Stephane de Labrusse <stephdl@de-labrusse.fr>
+ */
+class Rspamd extends \Nethgui\Module\AbstractModule implements \NethServer\Module\Dashboard\Interfaces\ApplicationInterface
+{
+
+    public function getName()
+    {
+        return "Rspamd";
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/Applications/Rspamd.php:33]: Show also the password (?)

File: root/etc/e-smith/db/configuration/defaults/rspamd/TCPPort [hunk 12]
```diff
@@ -0,0 +1 @@
+11334
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/defaults/rspamd/TCPPort]: No need to configure the firewall: the `TCPPort` prop can be removed as `rspamd` can be accessed from localhost only.

File: root/etc/e-smith/db/configuration/defaults/rspamd/rspamdWebUI [hunk 13]
```diff
@@ -0,0 +1 @@
+green
```
Comment (DavidePrincipi) [inline root/etc/e-smith/db/configuration/defaults/rspamd/rspamdWebUI]: Remove (?)

File: root/usr/share/nethesis/NethServer/Template/Mail/Filter.php [hunk 14]
```diff
@@ -58,10 +64,26 @@
...
+$db = $view->getModule()->getPlatform()->getDatabase('configuration');
+$app = $db->getProp('rspamd','Name');
+$password = $db->getProp('rspamd','password');
+
+//Retrieve the  rspamd URL
+$host = explode(':',$_SERVER['HTTP_HOST']);
+$url = "https://".$host[0]."/".$app."/";
+
+$webUI = $view->fieldset()->setAttribute('template', $T('Rspamd_WebUI_Settings_label'))
+    ->insert($view->radioButton('rspamdWebUI', 'red')->setAttribute('label', $T('rspamdWebUI_red_label')))
```
Comment (DavidePrincipi) [inline root/usr/share/nethesis/NethServer/Template/Mail/Filter.php]: Remove (?)

File: root/etc/e-smith/templates/etc/httpd/conf.d/default-virtualhost.inc/50Rspamd [hunk 15]
```diff
@@ -0,0 +1,7 @@
+
+#
+# 50Rspamd
+#
+RewriteEngine On
+RewriteCond %\{HTTPS\} !=on
+RewriteRule ^/{$rspamd{'Name'}}(/.*)?$ https://%\{HTTP_HOST\}/{$rspamd{'Name'}}$1 [L,R=301]
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/httpd/conf.d/default-virtualhost.inc/50Rspamd]: Remove fragment (?)

File: root/etc/e-smith/templates/etc/httpd/conf.d/rspamd.conf/00Allow [hunk 16]
```diff
@@ -0,0 +1,22 @@
+{
+use esmith::NetworksDB;
+
+my $ndb = esmith::NetworksDB->open_ro();
+my @localAccessSpec = $ndb->local_access_spec();
+
+if ($rspamd{rspamdWebUI} eq 'green'){
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/httpd/conf.d/rspamd.conf/00Allow]: Remove `root/etc/e-smith/templates/etc/httpd/conf.d/rspamd.conf` template (?)

**Review by DavidePrincipi — COMMENTED** [review]
Do you want to remove the rspamd/ UI configuration from the main web server instance? 

I'd keep it available on port 980 only:  as it's an admin interface I want it to be subject to the same policy of httpd-admin.

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**